QEfficient/transformers/models/falcon/modeling_falcon.py

```diff
@@ -0,0 +1,648 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# -----------------------------------------------------------------------------
+
+import math
+import warnings
+from typing import Optional, Tuple, Union
+
+import torch
+import torch.utils.checkpoint
+from torch import nn
+from torch.nn import CrossEntropyLoss, LayerNorm
+from torch.nn import functional as F
+from transformers.modeling_attn_mask_utils import AttentionMaskConverter
+from transformers.models.falcon.modeling_falcon import (
+    FalconAttention,
+    FalconConfig,
+    FalconDecoderLayer,
+    FalconForCausalLM,
+    FalconMLP,
+    FalconModel,
+    _prepare_4d_causal_attention_mask_for_sdpa,
+    build_alibi_tensor,
+    dropout_add,
+    logger,
+    rotate_half,
+)
+
+from QEfficient.transformers.modeling_attn_mask_utils import _qeff_prepare_4d_causal_attention_mask
+from QEfficient.transformers.modeling_outputs import (
+    QEffBaseModelOutputWithPastAndCrossAttentions,
+    QEffCausalLMOutputWithCrossAttentions,
+)
+
+
+def flip_torch(q, axis):
```

Comment:
This flip_torch function can be removed as it is not required for modeling file.